storybookjs/storybook
# Docs: Disable Markdown transclusion and support ?raw .md files

## What I did

This PR changes how plain `.md` files are handled and can be imported.

Previously Webpack (opt-in) and Vite could handle plain MD files using the MDX loader. However MDX2 got very strict with syntax, which results in perfectly valid `.md` files crashing the MDX loader, because it is invalid MDX syntax. This is confusing and have caused frustration. An example of a valid `.md` file that would break when imported is:

```md

{ this is perfectly valid markdown, but MDX treats this as an expression }

```

This behavior is being replaced with support for handling `?raw` imports. This means that it's now possible to import `.md` files as plain strings and do whatever necessary with them afterwards. The syntax for that is `import ReadMe from './README.md?raw';`

A follow-up PR will make this actually useful, by introducing a `Markdown` doc block that can handle these raw imports, like this:

```md
// example.mdx
import { Markdown } from '@storybook/blocks';
import ReadMe from './README.md?raw';

<Markdown>{ReadMe)</Markdown>
```

This will make it possible to import valid, plain MD into MDX files without issues.


1. Vite builder now doesn't use the `mdxLoader` to handle `.md` files, it is only used to handle `.mdx` files. This might be interesting to you @joshwooding as you originally built this in https://github.com/storybookjs/builder-vite/pull/429 cc @IanVS 
2. The `transcludeMarkdown` option has been removed. it was only implemented for Webpack, and it did the same as described above, except it was opt-in. Now it's not possible to opt-in to that anymore.

## How to test

- Create a sandbox with Webpack and with Vite
- import a `.md` file in an `.mdx` file with the raw tag, and see that it renders as a plain string without breaking.

Stories for the Markdown block will also test this in the follow-up PR.


## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
